### PR TITLE
[ENH] Implemented `NB_FEDERATE_REMOTE_PUBLIC_NODES` env var

### LINKED ISSUE #62: Add env var to disable federating over remote public nodes
- [x] f-API has ENV var that controls if public nodes are added to local nodes automatically
- [x] Figure out what the new default should be
- [x] Implement the new default

### PR DESCRIPTION
**Pull request by rmanaem**
- Closes #62 

## Checklist
_This section is for the PR reviewer_

- [ ] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see our [Contributing Guidelines](https://neurobagel.org/contributing/CONTRIBUTING#pull-request-guidelines) for more info)_
- [x] PR has a label for the release changelog or `skip-release` (to be applied by maintainers only)
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

**Review by surchs — COMMENTED** [review]
Thanks @rmanaem, this seems to do the trick. I do think we should 

- provide basic docs for this new feature. if README isn't the place, then please make a PR against the ENV var table in the main docs and link here
- add a basic integration test to make sure the ENV var does what we think it does
- unless I forgot, this is our first use of ENV vars in this tool. So let's add a `template.env` with the new variable and a short docstring explaining what it does.

File: app/api/utility.py [hunk 1]
```diff
@@ -2,14 +2,24 @@
...
 import httpx
 import jsonschema
 from fastapi import HTTPException, status
 from jsonschema import validate
 
+EnvVar = namedtuple("EnvVar", ["name", "value"])
+
+FEDERATE_REMOTE_PUBLIC_NODES = EnvVar(
+    "NB_FEDERATE_REMOTE_PUBLIC_NODES",
+    os.environ.get("NB_FEDERATE_REMOTE_PUBLIC_NODES", "True").lower()
```
[the remark this answers is not in the document]
Comment (rmanaem) [reply app/api/utility.py:19]: The hope was that using Python's `True` would prevent confusion (following the existing code format of the node API).

File: app/api/utility.py [hunk 2]
```diff
@@ -2,14 +2,24 @@
...
 import warnings
+from collections import namedtuple
 from pathlib import Path
 
 import httpx
 import jsonschema
 from fastapi import HTTPException, status
 from jsonschema import validate
 
+EnvVar = namedtuple("EnvVar", ["name", "value"])
```
[the remark this answers is not in the document]
Comment (rmanaem) [reply app/api/utility.py:15]: This is more to adhere to the existing code format of the node API and make it easier to add env vars (that may be used in multiple places) in the future.

**Comment by rmanaem** [comment]
> provide basic docs for this new feature. if README isn't the place, then please make a PR against the ENV var table in the main docs and link here

Addressing that here https://github.com/neurobagel/recipes/issues/60

> add a basic integration test to make sure the ENV var does what we think it does

Test has been added

File: app/api/utility.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (surchs) [reply app/api/utility.py:19]: ah OK, makes sense. let's keep it consistent then!

**Review by surchs — APPROVED** [review]
awesome, thanks @rmanaem. This looks good

:cook:

**Comment by surchs** [comment]
:rocket: PR was released in [`v0.2.0`](https://github.com/neurobagel/federation-api/releases/tag/v0.2.0) :rocket:

**Pull request merged**